enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/messaging/marshalling/v2/decoding/RaftMessageComposer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2002-2018 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j Enterprise Edition. The included source
+ * code can be redistributed and/or modified under the terms of the
+ * GNU AFFERO GENERAL PUBLIC LICENSE Version 3
+ * (http://www.fsf.org/licensing/licenses/agpl-3.0.html) with the
+ * Commons Clause, as found in the associated LICENSE.txt file.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * Neo4j object code can be licensed independently from the source
+ * under separate terms from the AGPL. Inquiries can be directed to:
+ * licensing@neo4j.com
+ *
+ * More information is also available at:
+ * https://neo4j.com/licensing/
+ */
+package org.neo4j.causalclustering.messaging.marshalling.v2.decoding;
+
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageDecoder;
+
+import java.time.Clock;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+import org.neo4j.causalclustering.core.consensus.RaftMessages;
+import org.neo4j.causalclustering.core.replication.ReplicatedContent;
+
+public class RaftMessageComposer extends MessageToMessageDecoder<Object>
+{
+    private final Queue<ReplicatedContent> replicatedContents = new LinkedBlockingQueue<>();
+    private final Queue<Long> raftLogEntryTerms = new LinkedBlockingQueue<>();
```

Comment:
Is there any reason of synchronisation why these need to be LinkedBlockingQueues? Otherwise I suggest some simple Lists. Remember that handlers in a netty pipeline do not run concurrently.

Reply:
No reason. The queue interface made sense to me because the decoding relies on FIFO for all the component. I was probably just lazy in picking this queue. Is there a better one in java?